File: app/src/main/java/com/example/polyquicktrade/ui/ProfileActivity.java
package com.example.polyquicktrade.ui;

import android.os.Bundle;

import com.bumptech.glide.Glide;
import com.example.polyquicktrade.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import de.hdodenhof.circleimageview.CircleImageView;

import android.util.Log;
import android.view.View;
import android.widget.TextView;

import java.util.Objects;

public class ProfileActivity extends AppCompatActivity {

    private static final String TAG = "MY_TAG";
    FirebaseAuth mFirebaseAuth;
    FirebaseUser firebaseUser;
    TextView nameTextview;
    TextView emailTextview;
    CircleImageView profileImageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(R.string.title_activity_profile);
        }
        mFirebaseAuth = FirebaseAuth.getInstance();

        nameTextview = findViewById(R.id.tv_username);
        emailTextview = findViewById(R.id.tv_email);

        profileImageView = findViewById(R.id.profile_image);


    }

    @Override
    protected void onStart() {
        super.onStart();

        firebaseUser = mFirebaseAuth.getCurrentUser();
        updateUI(firebaseUser);


    }

    private void updateUI(FirebaseUser firebaseUser) {

        if (firebaseUser != null) {

            nameTextview.setText(firebaseUser.getDisplayName());
            emailTextview.setText(firebaseUser.getEmail());



            Glide.with(this).load(firebaseUser.getPhotoUrl()).into(profileImageView);


        }
        else{
            //TODO handle error
        }

//        firebaseUser.getPhoneNumber();

    }


}

File: app/src/main/java/com/example/polyquicktrade/ui/ProfileActivity.kt
package com.example.polyquicktrade.ui

import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import android.widget.TextView
import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import com.example.polyquicktrade.R
import com.bumptech.glide.Glide
import de.hdodenhof.circleimageview.CircleImageView

class ProfileActivity : AppCompatActivity() {
    var mFirebaseAuth: FirebaseAuth? = null
    var firebaseUser: FirebaseUser? = null
    var nameTextview: TextView? = null
    var emailTextview: TextView? = null
    var profileImageView: CircleImageView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setTitle(R.string.title_activity_profile)
        }
        mFirebaseAuth = FirebaseAuth.getInstance()
        nameTextview = findViewById(R.id.tv_username)
        emailTextview = findViewById(R.id.tv_email)
        profileImageView = findViewById(R.id.profile_image)
    }

    override fun onStart() {
        super.onStart()
        firebaseUser = mFirebaseAuth!!.currentUser
        updateUI(firebaseUser)
    }

    private fun updateUI(firebaseUser: FirebaseUser?) {
        if (firebaseUser != null) {
            nameTextview!!.text = firebaseUser.displayName
            emailTextview!!.text = firebaseUser.email
            Glide.with(this).load(firebaseUser.photoUrl).into(profileImageView!!)
        } else {
            //TODO handle error
        }

//        firebaseUser.getPhoneNumber();
    }

    companion object {
        private const val TAG = "MY_TAG"
    }
}